### PR TITLE
fix ArrayFilter logic in IR

### PR DESCRIPTION
**Pull request by catoverdrive**

### CODE REVIEW

**Comment by catoverdrive** [comment]
@jbloom22 I realized that updating the length of the array after creating it doesn't work because of the variable size of the missingness bits that get allocated.

File: src/test/scala/is/hail/expr/ir/CompileSuite.scala [hunk 1]
```diff
@@ -467,4 +465,31 @@ class CompileSuite {
...
+  @Test
+  def testArrayFilterElement() {
+    val t = TArray(TInt32())
+    val ir = ArrayFilter(In(0, t), "x", ApplyBinaryPrimOp(EQ(), Ref("x"), In(1, TInt32())))
+    val region = Region()
+    val fb = FunctionBuilder.functionBuilder[Region, Long, Boolean, Int, Boolean, Long]
+    doit(ir, fb)
+    val f = fb.result(Some(new java.io.PrintWriter(System.out)))()
+    for {
+      stop <- 0 to 10
```
Comment (jbloom22) [inline src/test/scala/is/hail/expr/ir/CompileSuite.scala]: 5 is plenty

File: src/main/scala/is/hail/expr/ir/Emit.scala [hunk 2]
```diff
@@ -346,28 +350,27 @@ private class Emit(
...
+              Code(srvbKeep.addBoolean(true), lenout := lenout + 1)),
+            srvbKeep.advance(),
+            i := i + 1),
+          offKeep := srvbKeep.offset,
+          i := 0,
+          srvb.start(lenout, init = true),
+          Code.whileLoop(i < len,
+            region.loadBoolean(tKeep.loadElement(region, offKeep, i)).mux(
+              Code(
+                srvb.addIRIntermediate(t.elementType)(region.loadIRIntermediate(t.elementType)(t.loadElement(region, xa, i))),
```
Comment (jbloom22) [inline src/main/scala/is/hail/expr/ir/Emit.scala]: move up a line

File: src/main/scala/is/hail/expr/ir/Emit.scala [hunk 3]
```diff
@@ -346,28 +350,27 @@ private class Emit(
...
           i := 0,
           lenout := 0,
-          srvb.start(len, init = true),
+          srvbKeep.start(len, init = true),
           Code.whileLoop(i < len,
             xmv := !t.isElementDefined(region, xa, i),
-            xvv := xmv.mux(
-              defaultValue(t.elementType),
-              region.loadIRIntermediate(t.elementType)(t.loadElement(region, xa, i))),
+            xvv := xmv.mux(defaultValue(t.elementType), region.loadIRIntermediate(t.elementType)(t.loadElement(region, xa, i))),
```
Comment (jbloom22) [inline src/main/scala/is/hail/expr/ir/Emit.scala]: easier to read with region on next line

**Pull request merged**